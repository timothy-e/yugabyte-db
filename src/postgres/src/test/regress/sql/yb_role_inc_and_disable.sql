CREATE PROFILE ind_profile FAILED ATTEMPTS 2;
CREATE USER ind_user;

SELECT prfname, prffailedloginattempts FROM pg_catalog.pg_yb_profile ORDER BY OID;
SELECT rolname from pg_catalog.pg_roles WHERE rolname = 'ind_user';

ALTER USER ind_user PROFILE ATTACH ind_profile;

SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;

-- Simulate a failed attempt that increments the counter.
ALTER USER ind_user PROFILE ATTEMPTS FAILED;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;

 -- Simulate one more failure. This failure is allowed and does not disable the role.
ALTER USER ind_user PROFILE ATTEMPTS FAILED;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;

 -- Simulate one more failure. This attempt disables the role.
ALTER USER ind_user PROFILE ATTEMPTS FAILED;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;

-- Enable should reset the counter
ALTER USER ind_user PROFILE ENABLE;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;

ALTER USER ind_user PROFILE DETACH;
DROP USER ind_user;
DROP PROFILE ind_profile;
