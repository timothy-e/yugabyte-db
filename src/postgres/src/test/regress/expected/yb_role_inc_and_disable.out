CREATE PROFILE ind_profile LIMIT FAILED_LOGIN_ATTEMPTS 2;
CREATE USER ind_user;
SELECT prfname, prffailedloginattempts FROM pg_catalog.pg_yb_profile ORDER BY OID;
   prfname   | prffailedloginattempts
-------------+------------------------
 default     |                      0
 ind_profile |                      2
(2 rows)

SELECT rolname from pg_catalog.pg_roles WHERE rolname = 'ind_user';
 rolname
----------
 ind_user
(1 row)

ALTER USER ind_user PROFILE ATTACH ind_profile;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;
 rolisenabled | rolfailedloginattempts | rolname  |   prfname
--------------+------------------------+----------+-------------
 t            |                      0 | ind_user | ind_profile
(1 row)

-- Simulate a failed attempt that increments the counter.
ALTER USER ind_user PROFILE ADD FAILED_LOGIN_ATTEMPTS;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;
 rolisenabled | rolfailedloginattempts | rolname  |   prfname
--------------+------------------------+----------+-------------
 t            |                      1 | ind_user | ind_profile
(1 row)

 -- Simulate one more failure. This failure is allowed and does not disable the role.
ALTER USER ind_user PROFILE ADD FAILED_LOGIN_ATTEMPTS;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;
 rolisenabled | rolfailedloginattempts | rolname  |   prfname
--------------+------------------------+----------+-------------
 t            |                      2 | ind_user | ind_profile
(1 row)

 -- Simulate one more failure. This attempt disables the role.
ALTER USER ind_user PROFILE ADD FAILED_LOGIN_ATTEMPTS;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;
 rolisenabled | rolfailedloginattempts | rolname  |   prfname
--------------+------------------------+----------+-------------
 f            |                      3 | ind_user | ind_profile
(1 row)

-- Enable should reset the counter
ALTER USER ind_user PROFILE ENABLE;
SELECT rolisenabled, rolfailedloginattempts, rolname, prfname FROM
    pg_catalog.pg_yb_role_profile rp JOIN pg_catalog.pg_roles rol ON rp.rolid = rol.oid
    JOIN pg_catalog.pg_yb_profile lp ON rp.prfid = lp.oid;
 rolisenabled | rolfailedloginattempts | rolname  |   prfname
--------------+------------------------+----------+-------------
 t            |                      0 | ind_user | ind_profile
(1 row)

ALTER USER ind_user PROFILE DETACH;
DROP USER ind_user;
DROP PROFILE ind_profile;
